lib/functions.zsh

```diff
@@ -253,3 +253,30 @@ function omz_urldecode {
 
   echo -E "$decoded"
 }
+
+# The OS is linux
+function is_linux { [[ "$OSTYPE" == linux* ]]; }
+# The OS is Android
+function is_android { [[ "$OSTYPE" == linux-android* ]]; }
+# The OS is NetBSD
+function is_netbsd() { [[ "$OSTYPE" == netbsd* ]]; }
+# The OS is OpenBSD
+function is_openbsd() { [[ "$OSTYPE" == openbsd* ]]; }
+# The OS is FreeBSD
+function is_freebsd() { [[ "$OSTYPE" == freebsd* ]]; }
+# The OS is macOS (darwin)
+function is_mac() { [[ "$OSTYPE" == darwin* ]]; }
+# The OS is macOS (darwin) running on an M1 (arm64) CPU
```

Comment:
Could change this to be more generic. For Eg:

"The OS is macOS (darwin) running on series of ARM-based systems-on-a-chip designed by Apple Inc"

Reply:
Sure, no problem.

Reply:
Are these comments OK for you like they are now? @ParthJadhav 

```sh
# The OS is macOS (darwin) running on series of ARM-based systems-on-a-chip designed by Apple Inc
function is_mac_arm() { is_mac && [[ "$CPUTYPE" == arm64 ]]; }
# The OS is macOS (darwin) running on series of Intel (amd64 / x86_64) CPU
function is_mac_intel() { is_mac && [[ "$CPUTYPE" == x86_64 ]]; }
```

Reply:
Yes @casaper 👍